### PR TITLE
S3 actions for application

### PR DESCRIPTION
**Pull request by damogallagher**
This pull request contains the S3Service class and associated unit tests.
This provides functionality for uploading a file to S3, downloading a file from S3 and deleting a file from S3.
The unit tests for the service also exist.
There is a Rest Controller called S3Resource but this is only used for testing and will be removed when we have implemented this functionality in the angular application

This PR also contains a fix to the terraform to delete the S3 bucket on destrouy of the terraform stack if the bucket contains files

### CODE REVIEW

**Review by 97047619 — APPROVED** [review]
Approved

File: app/src/main/resources/config/application-prod.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@
 # Full reference is available at:
 # http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
 # ===================================================================
+aws:
+  region: eu-west-1
+  storageBucketName: test-bucket
```
Comment (L00163105) [inline app/src/main/resources/config/application-prod.yml:17]: Looks like the same bucket name for dev/prod

File: app/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (L00163105) [inline app/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Whats this do?

**Comment by damogallagher** [comment]
That is just being used for testing. The region and bucket are being
injected in as environment variables in elastic beanstalk and those values
are being used

**Comment by damogallagher** [comment]
Good question.this line magically allows you to mock final classes with
Mockito. Had to use it for 1 mock in the S3Service

**Review by L00163105 — APPROVED** [review]

**Review by JoseSunoj — APPROVED** [review]
Nice work. Approved.

**Pull request merged**